backend/grant/admin/views.py

```diff
@@ -712,3 +712,103 @@ def edit_comment(comment_id, hidden, reported):
+def financials():
+    grants = {
+        'total': '0',
+        'matching': '0',
+        'bounty': '0',
+    }
+    contributions = {
+        'total': '0',
+        'staking': '0',
+        'funding': '0',
+        'funded': '0',
+        'refunding': '0',
+        'refunded': '0',
+        'gross': '0',
+        'by_stage': {},
+    }
+    payouts = {
+        'total': '0',
+        'due': '0',
+        'paid': '0',
+        'future': '0',
+    }
+
+    def add_str_dec(a: str, b: str):
+        return str(Decimal(a) + Decimal(b))
+
+    def add_cont(amount, stage, is_refunded, is_staking):
+        contributions['total'] = add_str_dec(contributions['total'], amount)
+        if is_staking:
+            contributions['staking'] = add_str_dec(contributions['staking'], amount)
+            # staking only counts towards total & staking
+            return
+
+        if stage in contributions['by_stage']:
+            contributions['by_stage'][stage] = add_str_dec(contributions['by_stage'][stage], amount)
+        else:
+            contributions['by_stage'][stage] = add_str_dec('0', amount)
+
+        if stage in [ProposalStage.FUNDING_REQUIRED]:
+            contributions['funding'] = add_str_dec(contributions['funding'], amount)
+        if stage in [ProposalStage.COMPLETED, ProposalStage.WIP]:
+            contributions['funded'] = add_str_dec(contributions['funded'], amount)
+        if stage in [ProposalStage.CANCELED, ProposalStage.FAILED]:
+            if is_refunded:
+                contributions['refunded'] = add_str_dec(contributions['refunded'], amount)
+            else:
+                contributions['refunding'] = add_str_dec(contributions['refunding'], amount)
+
+    proposals = Proposal.query.all()
+
+    for p in proposals:
+        if p.status == ProposalStatus.LIVE:
+            if p.stage in [ProposalStage.WIP, ProposalStage.COMPLETED]:
+                # matching
+                matching = Decimal(p.contributed) * Decimal(p.contribution_matching)
+                remaining = Decimal(p.target) - Decimal(p.contributed)
+                if matching > remaining:
+                    matching = remaining
+
+                # bounty
+                bounty = Decimal(p.contribution_bounty)
+                remaining = Decimal(p.target) - (matching + Decimal(p.contributed))
+                if bounty > remaining:
+                    bounty = remaining
+
+                grants['matching'] = add_str_dec(grants['matching'], matching)
+                grants['bounty'] = add_str_dec(grants['bounty'], bounty)
+                grants['total'] = add_str_dec(grants['total'], matching + bounty)
+            
+            # contributions
+            for c in p.contributions:
+                if c.status == ContributionStatus.CONFIRMED:
+                    add_cont(c.amount, p.stage, True if c.refund_tx_id else False, c.staking)
+
+            # milestones/payouts
+            if p.stage in [ProposalStage.WIP, ProposalStage.COMPLETED]:
```

Comment:
Same deal, milestones for proposals in other stages still might have been paid out. This should probably be a direct query on milestones.